Create UserBook Controller

EwertonRosendo opened the pull request:
- ## Must create actions for UsersBookController(Review Controller)
- ### actions to create
- ### index 
  - will show all the data from the UserBook
- ### show
  - will show the book review by id
- ### destroy
  - will delete the book review
- ### edit 
  - will edit the book review
### Others changes

LucianoConceicaoJr commented on config/routes.rb:
```diff
@@ -12,8 +12,11 @@
 
-  get ":id/books", to: "users_book#index"
-  post ":id/books", to: "users_book#create"
+  get ":user_id/reviews", to: "users_book#index" # show all the users' reviews
```
@EwertonRosendo por que não usar resourceful routes?

EwertonRosendo committed:
create model Comment and its controller, change routes to use resources for some endpoints

Model Comment has 4 fields: user, review_id, content, likes. Routes: set 3 resources [users/reviews, users/comments, review/comments]

EwertonRosendo committed:
add has_many reviews for Book and has_many comments for User

LucianoConceicaoJunior commented on app/controllers/users_book_controller.rb:
```diff
@@ -1,23 +1,57 @@
+    @user_book = UsersBook.find(params[:id])
+  end
+
+  def params_hash
```
@EwertonRosendo qual a finalidade disso? Alias, tem certeza que funciona? Pois esses atributos não estão no permit(talvez funcione por conta dos IDs, mas não tenho certeza)

EwertonRosendo committed:
change routes.rb to use resources

EwertonRosendo committed:
small changes on routes.rb

EwertonRosendo committed:
fixing conflits

EwertonRosendo committed:
fixing prettier

EwertonRosendo committed:
change readme.md

EwertonRosendo committed:
remove conflits

EwertonRosendo committed:
remove wrong code from git merge

EwertonRosendo committed:
rewrite  js files and ruby models

EwertonRosendo committed:
merge with main
